### PR TITLE
add blog post on osquery sdk (Dec 21)

### PR DESCRIPTION
**Pull request by groob**
Looking forward to your reviews. 

// cc @zwass @marpaia

### CODE REVIEW

File: content/advent-2017/osquery-sdk.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+
+## Writing a custom logger plugin 
+
+When schedule a query like `SELECT name, version from deb_packages;` the `oqueryd` daemon will create a JSON log event with the results of the query. By default, a `filesystem` plugin is used, which logs the results to a local file. Commonly oquery users use aggregation tools like `filebeat`to send the result logs to a centralized log platform. Other plugins exist too. The `tls` plugin sends all logs to a remote TLS server like [Fleet](https://kolide.com/fleet). The `kinesis` plugin sends logs results to AWS, allowing advanced monitoring with applications like [StreamAlert](https://medium.com/airbnb-engineering/streamalert-real-time-data-analysis-and-alerting-e8619e3e5043). But what if you already have a well established logging pipeline with the systemd `journa`l, Splunk, `fluentd` or any number of proprietary logging systems. With the Thrift bindings to osquery, you can write your own logger. Go, having support for most APIs these days, is an ideal language for implementing a logger.
```
Comment (groob) [inline content/advent-2017/osquery-sdk.md]: When scheduling*

Comment (terracatta) [reply content/advent-2017/osquery-sdk.md]: `journa`l should be `journal`

Comment (object88) [reply content/advent-2017/osquery-sdk.md]: Should there be a space between "`filebeat`" and "to"?

File: content/advent-2017/osquery-sdk.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+
+    osquery> SELECT process.start_time, systemd.name AS service, process.name, listening.address, listening.port, process.pid FROM processes AS process JOIN listening_ports AS listening ON (process.pid = listening.pid) JOIN systemd ON systemd.pid = process.pid and listening.port = 443;
+    +------------+------------------+----------+---------+------+-------+
+    | start_time | service          | name     | address | port | pid   |
+    +------------+------------------+----------+---------+------+-------+
+    | 6308708    | nignx.service    | nginx    | ::      | 443  | 25859 |
+    +------------+------------------+----------+---------+------+-------+
+
+By configuring the query to run on a schedule, and using the logger plugin to aggregate the results centrally, we can begin to instrument our systems and create alerts.  Osquery allows for
```
Comment (object88) [inline content/advent-2017/osquery-sdk.md]: What does osquery allow for...?

**Pull request merged**